crates/sui-metrics-proxy/src/channels.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use axum::body::Bytes;
+use prometheus_parse::Scrape;
+use tokio::sync::mpsc::Receiver;
+use tracing::error;
+/// NodeMetric is a placeholder for metric data
+/// tbd in future PR
+#[derive(Debug)]
+pub struct NodeMetric {
+    pub host: String,
+    pub data: Bytes, // raw post data from node
+}
+
+/// An UpstreamConsumer accepts bytes from calling clients and is
+/// responsible for sending them on to upstream services to store
+/// the relayed metric
+pub struct UpstreamConsumer {
+    pub network: String,
+    pub receiver: Receiver<NodeMetric>,
+}
+
+impl UpstreamConsumer {
+    pub fn new(network: String, receiver: Receiver<NodeMetric>) -> Self {
+        Self { network, receiver }
+    }
+    pub async fn run(&mut self) {
+        while let Some(nm) = self.receiver.recv().await {
+            let network = self.network.to_owned();
+            tokio::spawn(async move {
+                let Ok(data) = std::str::from_utf8(&nm.data) else {
+                    error!("unable to decode bytes from relayed metrics");
+                    return;
+                };
+                let lines = data.lines().map(|s| Ok(s.to_owned()));
+                let mut metrics = Scrape::parse(lines).expect("sdf");
```

Comment:
you should use unwrap if the message you're passing to expect isn't helpful

Reply:
but if this is untrusted data we probably don't want to panic here and instead want to just skip processing the data, right?

Reply:
oups, that expect should not be there.  I meant to switch that to a match statement so we don't panic in there.